ingestion/src/metadata/ingestion/source/sql_source.py

```diff
@@ -71,6 +72,17 @@ def _get_table_description(schema: str, table: str, inspector: Inspector) -> str
     return description
 
 
+def _get_private_key_config(config: SQLConnectionConfig) -> SQLConnectionConfig:
+    new_config = copy.deepcopy(config)
+    if "private_key" in new_config.connect_args.keys() and type(
+        new_config.connect_args["private_key"] == bytes
+    ):
+        new_config.connect_args["private_key"] = str(
+            new_config.connect_args["private_key"]
+        )
+    return new_config
+
+
```

Comment:
We can optimize this method.
Use `GET` instead of "private_key" in new_config.connect_args.keys()
remove the type check as the users will send it in string.
return from inside the if loop if it has private key, and return original config at the end if it's not.

Reply:
Okay I'll make the changes